salt/utils/etcd_util.py

```diff
@@ -126,15 +130,165 @@ def __init__(
                 "protocol": "https",
             }
 
-        xargs = auth.copy()
-        xargs.update(certs)
+        self.xargs = auth.copy()
+        self.xargs.update(certs)
 
-        if HAS_LIBS:
-            self.client = etcd.Client(host, port, **xargs)
-        else:
-            raise CommandExecutionError(
-                "(unable to import etcd, module most likely not installed)"
-            )
+    def watch(self, key, recurse=False, timeout=0, index=None):
+        raise NotImplementedError()
+
+    def get(self, key, recurse=False):
+        """
+        Get the value of a specific key.  If recurse is true, defer to EtcdClient.tree() instead.
+        """
+        raise NotImplementedError()
+
+    def read(self, key, recursive=False, wait=False, timeout=None, waitIndex=None):
```

Comment:
For consistency, yes.  In order to follow the signature of the old API we exposed and avoid a deprecation cycle, unfortunately, I don't think so.  I can always throw it in as an extra kwarg, but it would be weird to have both `recursive` and `recurse` in the same function signature I think, although I may be alone on that.

Reply:
I think since we're going to go ahead and deprecate the v2 support, we should go ahead and take the time to clean up the API while we're at it.

Reply:
also from our discussion: `recurse` appears to be "more Salty" - `grep -r "recurse=" salt | wc -l` vs `grep -r "recursive=" salt | wc -l` - they're about the same number of occurrences, but the more core modules have a tendency to use `recurse`.

And generally, we may decide that it's not really worth keeping a consistent API across v2/v3, especially if the API doesn't make any sense with the way we're producing the values.